### PR TITLE
functions tasks are completed

### PR DESCRIPTION
**Pull request by emakhachek**

### CODE REVIEW

File: Exercises/functions/functions.js [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+const arrowFunction = (a, b, ...rest) => null;
+
+// Task #2
+function takeMyParametersAmount() {
+  let i = 0;
+  for (let arg of arguments)
+    i += 1;
+  console.log(i);
+  return null;
+}
```
Comment (Mike-Savin) [inline Exercises/functions/functions.js]: You can simplify it using `function.length` property:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length

File: Exercises/functions/functions.js [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+takeMyParametersAmount('a','b','c');
+
+// Task #3
+function logPassedParams() {
+  return arguments.length;
+}
```
Comment (Mike-Savin) [inline Exercises/functions/functions.js]: Rest syntax should be preferred:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters
Also you log not params, but only their length

File: Exercises/functions/functions.js [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+
+let count = logPassedParams(1, 2, 3);
+console.log(count);
+
+// Task #4
+(function () { })();
```
Comment (Mike-Savin) [inline Exercises/functions/functions.js]: Since your function is empty nothing is encapsulated

**Review by Mike-Savin — CHANGES_REQUESTED** [review]

File: Exercises/functions/functions.js [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+
+const functionExpression = function (a, b, ...rest) { return null; };
+
+const functionObject = new Function('a', 'b', 'return null');
+
+const arrowFunction = (a, b, ...rest) => null;
+
+// Task #2
+function takeMyParametersAmount(a, b, c) {
+  return console.log(takeMyParametersAmount.length);
```
Comment (lizarusi) [inline Exercises/functions/functions.js]: take length of the function itself, to get amount of parameters it expect

**Review by lizarusi — APPROVED** [review]
one small comment, gj :)

**Pull request merged**